oauth2/oauth2-jose/src/main/java/org/springframework/security/oauth2/jwt/JwtProcessors.java

```diff
@@ -159,4 +174,39 @@ public Resource retrieveResource(URL url) throws IOException {
 			}
 		}
 	}
+
+	/**
+	 * A builder for creating Nimbus {@link JWTProcessor} instances based on a
+	 * public key.
+	 */
+	public static final class PublicKeyJwtProcessorBuilder {
+		private RSAPublicKey key;
```

Comment:
Does it perhaps make sense to change this type to RSAKey right away since we don't need the RSAPublicKey type except for creating the RSAKey?